iree_tests/README.md

```diff
@@ -0,0 +1,111 @@
+# IREE Tests
+
+This directory contains generated test suites for running through IREE's
+compiler and runtime tools.
+
+Each test suite has one folder per test program containing a few files:
+
+```
+[program name 1]/
+  model.mlir
+  input_0.npy
+  output_0.npy
+  test_data_flags.txt
+```
+
+Where:
+
+* `model.mlir` is in a format that is ready for use with `iree-compile`
+  (e.g. torch-mlir, stablehlo, tosa, linalg)
+* `input_0.npy` and `output_0.npy` files correspond to any number of program
+  inputs and outputs for one test case
+* `test_data_flags.txt` is a flagfile for use with
+  `iree-run-module --flagfile=test_data_flags.txt` of the format:
+
+  ```
+  --input=@input_0.npy
+  --expected_output=@output_0.npy
+  ```
+
+## Running the test suite
+
+Developer quickstart:
+
+```bash
+python -m pip install iree-compiler iree-runtime
+python -m pip install -r ./iree_tests/onnx/requirements.txt
+
+python ./iree_tests/onnx/import_tests.py
+python ./iree_tests/compile_tests.py
+python ./iree_tests/run_tests.py
+```
+
+Testing follows several stages:
+
+1. Import
+2. Compile
+3. Run
```

Comment:
We'll also want a 'benchmark' stage for some programs / configurations. I'm not sure yet if that should chain off of the same test runner / framework, or just leverage the raw test suite files. Tightly coupling running benchmarks to building and testing in IREE's CI has been painful in some ways.